### PR TITLE
[UIKit] [Xaml] Add Custom Text Edit Sample to XAMLCatalog

### PR DESCRIPTION
**Pull request by davelamb**
Definitely should check the code out and run it. It's certainly a bit of crazy. :)

---

This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1257)

### CODE REVIEW

File: samples/XAMLCatalog/XAMLCatalog.vsimporter/XAMLCatalog-WinStore10/Package.appxmanifest [hunk 1]
```diff
@@ -7,15 +7,15 @@
...
     Publisher="CN=islandwood"
     Version="1.0.0.0" />
 
-  <mp:PhoneIdentity PhoneProductId="49ab0ab8-f42a-485e-9f46-c1348ba16d5c" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>
+  <mp:PhoneIdentity PhoneProductId="396fcbb6-2ce1-42ca-b99f-8a44ebc931a1" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>
 
   <Properties>
     <DisplayName>XAMLCatalog</DisplayName>
-    <PublisherDisplayName>pradipd</PublisherDisplayName>
+    <PublisherDisplayName>davelamb</PublisherDisplayName>
```
Comment (bbowman) [inline samples/XAMLCatalog/XAMLCatalog.vsimporter/XAMLCatalog-WinStore10/Package.appxmanifest]: I do sure wish there was something better we would do here. Its pretty silly that we don't just set this to  winobjc or something.

File: samples/XAMLCatalog/XAMLCatalog/CustomTextControlViewController.m [hunk 2]
```diff
@@ -0,0 +1,884 @@
...
+
+// This indicates that an IME has started composition.  If there is no handler for this event,
+// then composition will not start.
+- (void)_editContextCompositionStarted:(WUTCCoreTextEditContext*)sender args:(WUTCCoreTextCompositionStartedEventArgs*)e {
+}
+
+- (void)_editContextCompositionCompleted:(WUTCCoreTextEditContext*)sender args:(WUTCCoreTextCompositionCompletedEventArgs*)e {
+}
+
+- (void)dealloc {
```
Comment (oliversa-msft) [inline samples/XAMLCatalog/XAMLCatalog/CustomTextControlViewController.m]: Is it just me but I'd love to see the dealloc close to the init so I can compare the init/cleanup steps.

File: samples/XAMLCatalog/XAMLCatalog/CustomTextControlViewController.m [hunk 3]
```diff
@@ -0,0 +1,884 @@
...
+@end
+
+#ifdef WINOBJC
+// This class implements a custom edit control that is entirely
+// standalone. It does not integrate with XAML focus. Instead,
+// it manages focus explicitly by tracking pointer activity.
+
+// This view's only subview is an autoresized UIView that hosts
+// a Windows.UI.Xaml.ContentControl, which is used to gain focus,
+// tabstop, and recieve pointer events.
```
Comment (davelamb) [inline samples/XAMLCatalog/XAMLCatalog/CustomTextControlViewController.m]: 👍

File: samples/XAMLCatalog/XAMLCatalog/CustomTextControlViewController.m (hunk 2 above)
Comment (davelamb) [inline samples/XAMLCatalog/XAMLCatalog/CustomTextControlViewController.m]: As per coding spec. Will do. :)

**Comment by yiyang-msft** [comment]
:shipit:

**Pull request merged**